lib/Basic/LangOptions.cpp

```diff
@@ -391,6 +392,10 @@ void LangOptions::setHasAtomicBitWidth(llvm::Triple triple) {
     setMaxAtomicBitWidth(128);
     break;
 
+  case llvm::Triple::ArchType::avr:
+    setMaxAtomicBitWidth(0);
```

Comment:
Is this a "TODO, deal with this later" or is there really no atomics on AVR? Any details about this?

Reply:
if I recall correctly there are no atomic instructions, but you can mask interrupts do some ops and unmask